refactor: Light editing of server tuning guide (#13684)
nextcloud/documentation: admin_manual/installation/server_tuning.rst

Caching improves performance by storing data, code, and other objects in memory. Memory caching is not enabled by default because it requires optional extensions (such as APCu) and/or system components (e.g., Redis). Although these add-ons are generally not challenging to install and activate—at least in single-server deployments—you must install them before enabling their use in Nextcloud. See :doc:../configuration_server/caching_configuration for guidance.

Compression
-----------

Enabling compression in your web server for JavaScript, CSS, and SVG files improves performance 
because less data is transferred to clients.

Replacing SQLite
----------------

SQLite is a suitable database for some use cases, but using MariaDB, MySQL, or PostgreSQL can be 
more beneficial with Nextcloud.

If you do not select a database at installation time, SQLite is used by default because it does not require 
any external components.

However, MySQL/MariaDB or PostgreSQL are generally recommended for Nextcloud because of the 
`performance limitations of SQLite with highly concurrent applications
<https://www.sqlite.org/whentouse.html>`_, like Nextcloud.

If your installation is already running on

> SQLite, you can convert to MySQL or MariaDB using the steps provided in 

Author: ```suggestion
SQLite, you can convert to one of the suggested databases using the steps provided in 
```

.. danger::
    Increasing the time between revalidations (or disabling it completely) means that changes to scripts, including ``config.php``, will take longer to become active (or may never do so if revalidation is disabled completely). Increasing the interval also raises the risk of transient server and application upgrade problems and prevents the proper toggling of maintenance mode.

.. warning::
    If you adjust these parameters, you are more likely to need to restart/reload your web server (``mod_php``) or PHP-FPM after making configuration changes or performing upgrades. If you forget to do so, you may experience unusual behavior due to a mismatch between what is on disk and what is in memory. These may appear to be bugs, but will go away as soon as you restart/reload ``mod_php`` / fpm.

To change the default from ``2`` and check for changes on disk at most every ``60`` seconds, add the following setting to your ``php.ini`` file:

.. code:: ini

  opcache.revalidate_freq = 60

Alternatively, you can disable the revalidation completely:

.. code:: ini

  opcache.validate_timestamps = 0

> Any server or app upgrades, or changes to ``config.php``, will then require restarting PHP (or otherwise manually clearing the cache or invalidating this particular script).

Reviewer: Nextcloud and app upgrades trigger a full OPcache invalidation in the meantime. So this should apply only for cases where `config.php` is edited manually, config changes, app- or NC upgrades are done via CLI instead of web UI.

But this is probably something to address in a dedicated PR, which somehow makes the point clearer that the web UI and CLI use different OPcache instances, without encouraging readers too lightly to touch these settings. But anything done via web UI should never require a manual PHP restart/OPcache invalidation. Only changes done via CLI or to PHP files directly will do so, depending on (re)validation settings.

.. note::

>     Most Nextcloud instances use less than 2 MiB of the configured JIT buffer size, so 8 MiB is generally sufficient. 

Reviewer: Side note: In the meantime I found out that it depends on whether `tracing` JIT (or some profiling based trigger) is used, or whether scripts are just compiled unconditionally. This is the 3rd digit in `opcache.jit = 1255`: 5 being tracing (also default), and `0` means that everything is compiled directly on script load, in which case the JIT buffer usage can be about the same size as the regular opcode cache usage. With tracing JIT, the  `opcache.jit_buffer_size` seems to remain at a little above 1 MiB, most likely only containing some meta/profiling data, while the conditionally compiled scripts seems to be allocated from `opcache.memory_consumption` instead.

But something to address in a dedicated PR once I found time and mood to talk with some PHP developers about this, getting some clarification.

Author: Sounds good. I still haven't had an opportunity to get into the JIT stuff myself at all directly - other than what you've written/commented in the past. :)
